bears/vcs/VCSCommitBear.py

```diff
@@ -0,0 +1,149 @@
+import abc
+
+from aenum import Flag
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.misc.Shell import run_shell_command
+from coalib.results.HiddenResult import HiddenResult
+from coala_utils.decorators import (enforce_signature, generate_ordering,
+                                    generate_repr)
+
+
+class COMMIT_TYPE(Flag):
+    simple_commit = 0
+    merge_commit = 1
+
+
+@generate_repr(('id', hex),
+               'origin',
+               'raw_commit_message',
+               'commit_sha',
+               'parent_commits',
+               'commit_type',
+               'modified_files',
+               'added_files',
+               'deleted_files')
+@generate_ordering('raw_commit_message',
+                   'commit_sha',
+                   'parent_commits',
+                   'commit_type',
+                   'modified_files',
+                   'added_files',
+                   'deleted_files',
+                   'severity',
+                   'confidence',
+                   'origin',
+                   'message_base',
+                   'message_arguments',
+                   'aspect',
+                   'additional_info',
+                   'debug_msg')
+class CommitResult(HiddenResult):
+
+    @enforce_signature
+    def __init__(self, origin,
+                 raw_commit_message: str,
+                 commit_sha: str,
+                 parent_commits: list,
+                 commit_type: COMMIT_TYPE,
+                 modified_files: list,
+                 added_files: list,
+                 deleted_files: list):
+
+        super().__init__(origin, '')
+        self.raw_commit_message = raw_commit_message
+        self.commit_sha = commit_sha
+        self.parent_commits = parent_commits
+        self.commit_type = commit_type
+        self.modified_files = modified_files
+        self.added_files = added_files
+        self.deleted_files = deleted_files
+
+
+class VCSCommitBear(GlobalBear):
+    __metaclass__ = abc.ABCMeta
+    LANGUAGES = {'Git'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    @abc.abstractmethod
+    def get_head_commit(self):
+        """
+        Return the commit message from the head commit.
+        """
+
+    def analyze_git_commit(self, head_commit):
```

Comment:
remove "git" from this name, as not all vcs are git, and it should be abstract.

Then move git implementation to a new subclass GitCommitMetadataBear